Copy the correct link from the contribution page and insert it on the Recent Spectra Section for the contributor's name that currently has a broken link

Georjane opened the pull request:
Fixes #(update with issue number)

Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] tests pass -- `rake test`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request are descriptively named
* [x] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

Thanks!

Georjane commented:
> This looks great. I assume you tested it manually but do you see any opportunity for a test? If not, shall we merge this? Thanks!!!

Sure Jeff, I can definitely try to write a test for this. Thanks

Tlazypanda reviewed (commented):
Hey @Georjane for writing test you can checkout the system test in spectral-workbench and plots2 repo for reference. 
https://github.com/publiclab/spectral-workbench/blob/main/test/system/upload_test.rb
https://github.com/publiclab/plots2/blob/main/test/system/dashboard_test.rb

The idea would be to navigate to the recent spectral page link and then find the element on this page, click on it and check for redirect

Georjane commented:
> Hey @Georjane for writing test you can checkout the system test in spectral-workbench and plots2 repo for reference. https://github.com/publiclab/spectral-workbench/blob/main/test/system/upload_test.rb https://github.com/publiclab/plots2/blob/main/test/system/dashboard_test.rb
> 
> The idea would be to navigate to the recent spectral page link and then find the element on this page, click on it and check for redirect

Thanks very much @Tlazypanda This is very helpful

Tlazypanda commented:
Hey @Georjane can you please clear the merge conflicts through the command line so the tests can be run and then we can debug this? 

Also sharing this link for reference on how to remove merge conflicts through the command line if needed - https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line

Georjane commented:
Hi @jywarren @Tlazypanda @TildaDares 👋🏽 
I think I fixed the link for the spectrum author and I added the tests. But I have a question while the format that was in the code `profile_path(@user)` was not working while `/profile/user_id` works. See the two codes below.
##### working code
`<a class="author-profile"  data-transition="slide" href="/profile/<%=h spectrum.author %>"><%= spectrum.author %></a>` 
##### code not working
`<a class="author-profile" data-transition="slide" href="<%= profile_path(spectrum.author) %>"><%= spectrum.author %></a>`

TildaDares commented:
@Georjane Since the route for `profile` was not created with a resourceful route (`resources :profile`), helper methods like `_url` and `path` are not available for use. You can read more about it in the [Rails docs](https://guides.rubyonrails.org/routing.html#path-and-url-helpers). 

You can also use route helpers with routes like 
https://github.com/publiclab/spectral-workbench/blob/7291df72288a41d080ac91d66d66154435b749b0/config/routes.rb#L29
 by using `as` https://guides.rubyonrails.org/routing.html#naming-routes

Georjane commented:
> @Georjane Since the route for `profile` was not created with a resourceful route (`resources :profile`), helper methods like `_url` and `path` are not available for use. You can read more about it in the [Rails docs](https://guides.rubyonrails.org/routing.html#path-and-url-helpers).
> 
> You can also use route helpers with routes like
> 
> https://github.com/publiclab/spectral-workbench/blob/7291df72288a41d080ac91d66d66154435b749b0/config/routes.rb#L29
> 
> 
> by using `as` https://guides.rubyonrails.org/routing.html#naming-routes
> I hope that helps.

woow this is very helpful @TildaDares Thank you very much.
This means `get '/profile/:id', to: 'users#show' as: :profile` will create a route`profile_path` and this will solve the problem of the author broken link right?

Georjane commented:
@TildaDares I tried creating a fto for this issue, please can you check it out here, https://github.com/publiclab/spectral-workbench/issues/869
thank you! 🙏🏽

jywarren commented:
Oh cool - so does that mean after #869 is solved, we should be able to rebase this and it should work? Thanks!!
